### PR TITLE
Restyle feature boxes on main page and ready for vanilla Docsy

### PR DESCRIPTION
**Pull request by sftim**
Change how we put the feature boxes on the main page.

- use CSS for layout

Here's a [preview](https://deploy-preview-48308--kubernetes-io-main-staging.netlify.app/)

Helps with issue https://github.com/kubernetes/website/issues/41171

/area web-development

### CODE REVIEW

File: assets/scss/_base.scss [hunk 1]
```diff
@@ -687,13 +687,56 @@ section#cncf {
...
+    max-width: clamp(75em, 25cm, 90vw);
+    margin-left: auto;
+    margin-right: auto;
+
+    background: initial;
+
+    & > .feature-box {
+        h4 {
+          line-height: normal;
+          margin-bottom: 1.9em;
```
Comment (dipesh-rawat) [inline assets/scss/_base.scss:723]: There is a bit of extra spacing between the heading and content within the div. Reducing it slightly could make it look better.
```suggestion
          margin-bottom: 1em;
```

File: assets/scss/_base.scss [hunk 2]
```diff
@@ -687,13 +687,56 @@ section#cncf {
...
+        }
```
Comment (dipesh-rawat) [inline assets/scss/_base.scss:724]: Looks like these lines of CSS are targeting the same element as the later lines in the file. Are they redundant?

https://github.com/kubernetes/website/blob/8924f40f2d004a25adfbd0bc0496a45648a263d0/assets/scss/_base.scss#L708-L711

**Review by dipesh-rawat — COMMENTED** [review]
As part of these changes, should we also remove the following CSS rules? They might not be needed.

https://github.com/kubernetes/website/blob/8924f40f2d004a25adfbd0bc0496a45648a263d0/assets/scss/_base.scss#L720-L733

File: assets/scss/_base.scss
Comment (dipesh-rawat) [inline assets/scss/_base.scss:1]: Possibly, the below CSS lines can also be removed?

https://github.com/kubernetes/website/blob/8924f40f2d004a25adfbd0bc0496a45648a263d0/assets/scss/_base.scss#L699-L718

File: assets/scss/_base.scss (hunk 2 above)
Comment (sftim) [reply assets/scss/_base.scss]: I dropped them. Thanks.

File: assets/scss/_base.scss (hunk 1 above)
Comment (sftim) [reply assets/scss/_base.scss]: I redid the margins and padding here.

File: assets/scss/_base.scss
Comment (sftim) [reply assets/scss/_base.scss:1]: Removed.

**Review by dipesh-rawat — COMMENTED** [review]
/lgtm

**Comment by sftim** [comment]
(not yet ready to merge; I know there are snags, so)
/lgtm cancel

**Comment by sftim** [comment]
D'oh! Wrong PR.

@dipesh-rawat would you review again? Sorry.

**Review by dipesh-rawat — COMMENTED** [review]
/lgtm

**Pull request merged**